File: BST/BSTtoLL.java
package BST;    
import java.util.*;

class BSTtoLL{

    static class Node{
        int data;
        Node left;
        Node right;

        Node(int data){
            this.data = data;
            this.left = null;
            this.right = null;
        }
    }

    static class BST{
        Node root = null;
        Node head = null;
        Node tail = null;

        Node insert(Node root, int val){
            if (root == null){
                root = new Node(val);
            }

            else if( val < root.data){
                root.left = insert(root.left, val);
            }

            else if( val > root.data){
                root.right = insert(root.right, val);
            }

            return root;
        }

        void toLL(Node cur){
            if(cur == null){
                return;
            }

            toLL(cur.left);

            if(head == null){
                head = cur;
            }
            else{
                cur.right = tail;
                tail.left = cur;
            }
        }

        void inorder(Node root){
            if (root == null){
                return;
            }

            inorder(root.left);
            System.out.print(root.data + " ");
            inorder(root.right);
        }
    }

    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        BST tree = new BST();
        int n = sc.nextInt();

        for (int i = 0; i < n ; i++){
            int data = sc.nextInt();
            tree.root = tree.insert(tree.root, data);
        }

        tree.inorder(tree.root);

        sc.close();

    }
}

File: BST/BST_To_LL.py
class Node:
    def __init__(self) -> None:
        self.data = None
        self.left = None
        self.right = None

class BST:
    def __call__(self):
        self.root = None

    def insert(self, root, val):

        if self.root is None:
            self.root = Node(val)

        elif root is None:
            root = Node(val)

        elif (val < root.data):
            self.insert(root.left, val)

        else:
            self.insert(root.right, val)

    def inorder(self, root):

        if root is None:
            return None
        
        self.inorder(root.left)

        print(root.data)

        self.inorder(root.right)
        
if __name__ == '__init__':
    Bst = BST()

    nums = [14,10,8,64,9]
    
    for num in nums:
        Bst.insert(Bst.root, num)

    Bst.inorder(Bst.root)
